source/common/formatter/substitution_formatter.cc

```diff
@@ -129,37 +124,61 @@ std::string JsonFormatterImpl::format(const Http::RequestHeaderMap& request_head
   return absl::StrCat(log_line, "\n");
 }
 
+JsonFormatterImpl::JsonFormatMap
+JsonFormatterImpl::toFormatMap(const ProtobufWkt::Struct& json_format) const {
+  auto output = std::make_unique<std::map<std::string, JsonFormatterImpl::JsonFormatMapValue>>();
```

Comment:
I assume we want to preserve order here? Mind to add a comment here so we understand why we choose `std::map`.

Reply:
Done. Given that I've made aliases per the other comment, the explanation comment is now in the header file where the type's defined.